src/tools/miri/src/shims/unix/foreign_items.rs

```diff
@@ -967,6 +966,28 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let result = this.nanosleep(req, rem)?;
                 this.write_scalar(result, dest)?;
             }
+            "clock_nanosleep" => {
+                // Currently this function does not exist on all Unixes, e.g. on macOS.
+                this.check_target_os(
+                    &[
+                        "freebsd",
+                        "netbsd",
+                        "linux",
+                        "android",
+                        "solaris",
+                        "illumos",
+                        "dragonfly",
+                        "hurd",
+                        "fuchsia",
+                        "vxworks",
```

Comment:
We don't support most of these in Miri anyway, so let's cut this down to what makes sense.
```suggestion
                        "freebsd",
                        "linux",
                        "android",
                        "solaris",
                        "illumos",
```